packages/salesforce-adapter/src/filters/formula_deps.ts

```diff
@@ -76,14 +73,14 @@ const referencesFromIdentifiers = async (typeInfos: FormulaIdentifierInfo[]): Pr
 
 const addDependenciesAnnotation = async (field: Field, allElements: ReadOnlyElementsSource): Promise<void> => {
   const isValidReference = async (elemId: ElemID): Promise<boolean> => {
-    if (elemId.idType === 'type' || elemId.idType === 'instance') {
-      return await allElements.get(elemId) !== undefined
+    if (elemId.idType === 'field') {
+      const typeElemId = new ElemID(elemId.adapter, elemId.typeName)
+      const typeElement = await allElements.get(typeElemId)
+      return (typeElement !== undefined) && (typeElement.fields[elemId.name] !== undefined)
     }
 
-    // field
-    const typeElemId = new ElemID(elemId.adapter, elemId.typeName)
-    const typeElement = await allElements.get(typeElemId)
-    return (typeElement !== undefined) && (typeElement.fields[elemId.name] !== undefined)
+    // 'type' or 'instance'
+    return await allElements.get(elemId) !== undefined
```

Comment:
> In the previous code there was an assumption that any id which is not `type` or `instance` is a field but that is not true, it is not safe to make that assumption

How so? We define the typeId in `identifierTypeToElemIdType`, no?

Reply:
this is a function that gets `ElemID`, how is that related to `identifierTypeToElemIdType`?

also, no need to read the content of the element just to check if it exists
```suggestion
    // 'type' or 'instance'
    return allElements.has(elemId)
```